Makefile

```diff
@@ -79,6 +87,9 @@ LIBDM_BUILD_TAG = $(shell hack/libdm_tag.sh)
 LOCAL_BUILD_TAGS = $(BTRFS_BUILD_TAG) $(LIBDM_BUILD_TAG)
 BUILDTAGS += $(LOCAL_BUILD_TAGS)
 
+# When running inside the CI/Dev container, all tags are known.
+CONTAINER_BUILDTAGS ?= btrfs_noversion libdm_no_deferred_remove
```

Comment:
> Is this just to avoid running the hack/ autodetection scripts inside the container’s make?

Yes, I tried doing that and got it working, but wasn't happy that it would take up a few extra seconds of precious developer time.  The other thing I found was the scripts were ultimately resulting in an empty `BUILDTAGS` (maybe that's correct behavior?).

In any case, the contents of the container are entirely known and under our control, so I thought sticking in a default made sense.

> it might be a a bit surprising

Yeah, that's a good point, we don't want that :frowning: 

Maybe my fear of causing confusion between host and container BUILDTAGS is unfounded?

I'm open to other ideas/suggestions to make this easier, if you have any.

Reply:
> The other thing I found was the scripts were ultimately resulting in an empty BUILDTAGS (maybe that's correct behavior?).

I think so, the tags disable missing components or work around missing API.  But then `CONTAINER_BUILDTAGS` should be empty, to match the outcome, shouldn’t it?

---

Isn’t the problem here is that we mix user-input BUILDTAGS with the autodetected values with BUILDTAGS += LOCAL_TAGS? Without that, we could just forward the user choice directly into the container, and let both the outside and inside environment choose LOCAL_TAGS as appropriate (e.g. no autodetection in a container). But fixing that feels like a good topic for a separate PR, not something this one should deal with…

---

A reasonably cheap hack might be to skip the LOCAL_TAGS auto detection if SKOPEO_CONTAINER_TESTS == 1; that way we know we are in a known environment, we avoid running the slow shell script, all _extra_ user overrides via BUILDTAGS still work. The downside is that out-of-container LOCAL_TAGS still leak inside.

Reply:
Right, since when `make` is re-executed inside the container, it should do the right thing w/ running the scripts (and they're quick).  Okay, so I think this is a non-problem and It's safe for me to "un-fix" it :grinning:

Reply:
I've stripped it out, and also removed all the passing-in of `BUILDTAGS` when **not** running in a container (so there is no leaking).  We/Me/Someone can open a separate PR to add in something like 'EXTRA_BUILDTAGS' to give developers some control, when running the container-based targets.